### DISCUSSION TITLE
How to disabled the Modal footer button according to the form validate status in the Modal

### DISCUSSION THREAD

**Discussion by shanDongLiuyy** [ant-design/ant-design#30721] [Q&A]
`const [form] = Form.useForm();
<Modal
title="添加配方"
centered
visible={visible}
okButtonProps={{
  disabled: !form.isFieldsTouched(['menuName', 'sugar', 'temperature', 'materials']) ||
  !!form.getFieldsError().filter(({ errors }) => errors.length).length
}}
>
<Form>
<Form.Item
name="menuName" 
label="品名" 
rules={[
{
required: true,
message: '请输入品名'
}
]}
>
<Input />
</Form.Item>
</Form>
</Modal>`

When user open the Modal, what I want is the footer button is disabled, and after user enter something valid,then the button is enabled automatically

Thanks

**Comment by wodka** [comment]
you will need to wrap the modal in a form field that actually updates like: 

```tsx
<Form.Item shouldUpdate>
  {() => (
    <Modal 
      okButtonProps={{
        disabled: !form.isFieldsTouched(['menuName', 'sugar', 'temperature', 'materials']) ||
!!form.getFieldsError().filter(({ errors }) => errors.length).length
      }}
    >
      your body
    </Modal>
  }
</Form.Item>
```
